test/rpc_test.py

```diff
@@ -1193,17 +1177,20 @@ def test_local_shutdown_with_rpc(self):
             world_size=self.world_size,
             rpc_backend_options=self.rpc_backend_options,
         )
+        rpc._set_rpc_timeout(timedelta(milliseconds=100))
         n = self.rank + 1
         dst_rank = n % self.world_size
-        ret = rpc.rpc_sync(
-            "worker{}".format(dst_rank),
-            torch.add,
-            args=(torch.ones(n, n), torch.ones(n, n)),
-        )
-        # wait for RPCs to be done, so that some workers don't try to shut down
-        # too early.
-        rpc.rpc_sync("worker{}".format(dst_rank), _set_rpc_done, args=(1,))
-        _check_rpc_done(1)
+        # this RPC may timeout, if the dest worker has terminated before sending
+        # the response. A local shutdown does not guarantee that all outstanding
+        #  RPCs will be completed, so we catch timeout errors here.
+        try:
+            rpc.rpc_sync(
+                "worker{}".format(dst_rank),
+                torch.add,
+                args=(torch.ones(n, n), torch.ones(n, n)),
+            )
+        except RuntimeError as e:
+            self.assertTrue("RPC ran for more than" in str(e))
         # pass in graceful=False to ensure that we don't wait for other workers.
```

Comment:
Can't we just add a `dist.barrier` before the shutdown to fix the problem?

Reply:
Good idea, I think this is a better solution.